accounting-service/app/src/main/java/com/microservices/accounting/service/PaymentServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.microservices.accounting.service;
+
+import com.microservices.accounting.dto.InvokePaymentDto;
+import com.microservices.accounting.dto.PaymentDetailsDto;
+import com.microservices.accounting.dto.PaymentStatus;
+import com.microservices.accounting.dto.RevertPaymentDto;
+import com.microservices.accounting.entity.PaymentEntity;
+import com.microservices.accounting.repository.PaymentRepository;
+import com.microservices.accounting.temporaryclasses.CardInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.servlet.UnavailableException;
+import java.util.Objects;
+import java.util.Random;
+
+@Service
+public class PaymentServiceImpl implements PaymentService {
+    private PaymentRepository paymentRepository;
+    private final Logger logger = LoggerFactory.getLogger(PaymentServiceImpl.class);
+
+    @Autowired
+    public PaymentServiceImpl(PaymentRepository paymentRepository) {
+        this.paymentRepository = paymentRepository;
+    }
+
+    @Override
+    public PaymentDetailsDto invokePayment(InvokePaymentDto invokePaymentDto) {
+        Objects.requireNonNull(invokePaymentDto);
+
+        logger.info("Invoking payment: {}...", invokePaymentDto);
+        CardInfo cardInfo = invokePaymentDto.getUser().getCardInfo();
+        PaymentStatus paymentStatus = (cardInfo == CardInfo.VALID && isRequestApproved())
+                ? PaymentStatus.ACCEPTED
+                : PaymentStatus.DENIED;
+
+        PaymentEntity savedPayment = paymentRepository.save(new PaymentEntity(invokePaymentDto, paymentStatus));
+        logger.info("Saved payment: {}", savedPayment);
```

Comment:
Completed or Invoked ot Performed or I don't know which word more suitable for payment